lib/kitchen/driver/ec2.rb

```diff
@@ -289,7 +289,8 @@ def default_ami
       def update_username(state)
         # TODO: if the user explicitly specified the transport's default username,
         # do NOT overwrite it!
-        if instance.transport[:username] == instance.transport.class.defaults[:username]
+        if actual_platform &&
+            instance.transport[:username] == instance.transport.class.defaults[:username]
```

Comment:
the bug is that if the user tried to set the username to the default where the username is not nil by default, then it'll go off and do the autodetection and may choose to override the default.  this check for username != default cannot catch that case -- unless the default value is something like nil which the user would never set.

it feels like someone was thinking too hard about default values here.